File: tester/src/tests/media.ts
import axios from "axios";
import * as Scc from '../scc';
import * as Utils from '../utils';
import { TestEnv, Tester, TestNamespace } from "../tester";

export function register(): TestNamespace {
    const ns = new TestNamespace('media');
    ns.register('upload-image', uploadImageTest);
    ns.register('upload-download-image', uploadDownloadImageTest);
    ns.register('download-missing-image', downloadMissingImage);
    return ns;
}

async function uploadImageTest(env: TestEnv) {
    const requests = new Scc.Requests(env.endpoints);
    const response = await requests.uploadMedia(Scc.randomUploadMediaParams());
    env.validate(response, (response, assertions) => {
        assertions.status(response.status, 200);
        assertions.true(response.data !== undefined, 'Response data is undefined');
        assertions.true((response.data as string).length > 0, 'Response data is empty');
    });
}

async function uploadDownloadImageTest(env: TestEnv) {
    const requests = new Scc.Requests(env.endpoints);
    const client = new Scc.Client(env.endpoints);

    const uploadParams = Scc.randomUploadMediaParams();
    const mediaId = await client.uploadMedia(uploadParams);
    const response = await requests.downloadMedia(mediaId);
    env.validate(response, (response, assertions) => {
        assertions.status(response.status, 200);
        assertions.true(response.data !== undefined, 'Response data is undefined');

        const responseData = response.data as ArrayBuffer;
        assertions.true(responseData.byteLength > 0, 'Response data is empty');
        assertions.true(Utils.compareArrayBuffers(responseData, uploadParams), 'Response data is not equal to uploaded data');
    });
}

async function downloadMissingImage(env: TestEnv) {
    const requests = new Scc.Requests(env.endpoints);
    const response = await requests.downloadMedia('missing-media-id');
    env.validate(response, (response, assertions) => {
        assertions.status(response.status, 404);
    });
}

File: tester2/tests/media.py
import recon
import scc


def group(endpoints: scc.Endpoints):
    return recon.TestGroup(
        [
            upload_image(endpoints),
            upload_image_and_verify(endpoints),
            download_missing_image(endpoints),
        ]
    )


@recon.test_case("media", "upload image")
def upload_image(endpoints: scc.Endpoints):
    rclient = scc.RawClient(endpoints)
    response = rclient.upload_media(scc.random_image())
    with recon.validate(response) as validator:
        validator.status_code(200)


@recon.test_case("media", "upload image and verify download")
def upload_image_and_verify(endpoints: scc.Endpoints):
    rclient = scc.RawClient(endpoints)
    client = scc.Client(endpoints)

    image = scc.random_image()
    media_id = client.upload_media(image)
    response = rclient.download_media(media_id)
    with recon.validate(response) as validator:
        validator.status_code(200)
        validator.content_type("application/octet-stream")
        validator.content(image)


@recon.test_case("media", "download missing image")
def download_missing_image(endpoints: scc.Endpoints):
    rclient = scc.RawClient(endpoints)
    response = rclient.download_media("missing")
    with recon.validate(response) as validator:
        validator.status_code(404)
